node/src/components/consensus/protocols/highway.rs

```diff
@@ -403,11 +404,32 @@ where
                         )];
                     }
                 };
+                let is_faulty = match pvv.inner().signed_wire_unit() {
+                    Some(signed_wire_unit) => self
+                        .highway
+                        .state()
+                        .is_faulty(signed_wire_unit.wire_unit.creator),
+                    None => false,
+                };
+
                 match pvv.timestamp() {
                     Some(timestamp) if timestamp > Timestamp::now() => {
-                        self.store_vertex_for_addition_later(timestamp, sender, pvv)
+                        // If it's not from an equivocator and from the future, add to queue
+                        if !is_faulty {
+                            self.store_vertex_for_addition_later(timestamp, sender, pvv)
+                        } else {
+                            vec![]
+                        }
```

Comment:
Matter of taste but I find conditional statements without negations easier. So this could be _if by equivocator, don't store_. Up to you of course.

Reply:
I did it to be consistent with line 427; however in general I have the same preference